app/models/apple/episode.rb

```diff
@@ -0,0 +1,273 @@
+# frozen_string_literal: true
+
+require 'uri'
+
+module Apple
+  class Episode
+    attr_accessor :show, :feeder_episode, :api
+
+    def self.get_episodes(api, episodes)
+      return if episodes.empty?
+
+      api.bridge_remote_and_retry!('getEpisodes', episodes.map(&:get_episode_bridge_params))
+    end
+
+    def self.create_episodes(api, episodes)
+      return if episodes.empty?
+
+      episode_bridge_results = api.bridge_remote_and_retry!('createEpisodes',
+                                                            episodes.map(&:create_episode_bridge_params))
+
+      insert_sync_logs(episodes, episode_bridge_results)
+    end
+
+    def self.update_episodes(api, episodes)
+      return if episodes.empty?
+
+      episode_bridge_results = api.bridge_remote_and_retry!('updateEpisodes',
+                                                            episodes.map(&:update_episode_bridge_params))
+      insert_sync_logs(episodes, episode_bridge_results)
+    end
+
+    def self.update_audio_container_reference(api, episodes)
+      return [] if episodes.empty?
+
+      episode_bridge_results = api.bridge_remote_and_retry!('updateEpisodes',
+                                                            episodes.map(&:update_episode_audio_container_bridge_params))
+      insert_sync_logs(episodes, episode_bridge_results)
+    end
+
+    def self.publish(api, episodes)
+      return [] if episodes.empty?
+
+      api.bridge_remote_and_retry!('publishEpisodes',
+                                   episodes.map(&:publish_episode_bridge_params))
+    end
+
+    def self.insert_sync_logs(episodes, results)
+      episodes_by_item_guid = episodes.map { |ep| [ep.item_guid, ep] }.to_h
+
+      results.map do |res|
+        apple_id = res.dig('api_response', 'val', 'data', 'id')
+        guid = res.dig('api_response', 'val', 'data', 'attributes', 'guid')
+        ep = episodes_by_item_guid.fetch(guid)
+
+        SyncLog.
+          create(feeder_id: ep.feeder_episode.id, feeder_type: :episodes, external_id: apple_id)
+      end
+    end
+
+    def self.add_no_imp_param(url)
+      # use the URI class to add a 'noImp' query param
+      url = URI.parse(url)
+      decoded_query = URI.decode_www_form(url.query.to_s) << ['noImp', '1']
```

Comment:
Ah, ye olde `?noImp`.

Alternatively, could we set a bot-ish UA when making requests to Dovetail?  That's probably a good idea anyways, so we can tell where traffic is coming from.  And there's prior art for that in the podcast world (bots/scrapers like the Spotify one).

Then we can mark the UA as a "bot" in prx-podagent.  And get `is_duplicate=true` records in BQ, so we can debug what's going out to Apple.